File: samples/Linux/netcore_for_Linux_x64_only/netcore1.0/TableManagerSamples/PrintTableListener/src/TableManagerListener.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;
using fxcore2;

namespace PrintTableListener
{
    class TableManagerListener : IO2GTableManagerListener
    {
        private EventWaitHandle mEventWaitHandler;
        O2GTableManagerStatus mLastStatus;
        private bool mLoaded;
        private bool mError;

        public bool IsLoaded
        {
            get { return mLoaded; }
        }

        public bool HasError
        {
            get { return mError; }
        }

        // ctor
        public TableManagerListener()
        {
            Reset();
            mEventWaitHandler = new EventWaitHandle(false, EventResetMode.AutoReset);
        }

        public void Reset()
        {
            mLastStatus = O2GTableManagerStatus.TablesLoading;
            mLoaded = false;
            mError = false;
        }

        public bool WaitEvents()
        {
            if (mLastStatus == O2GTableManagerStatus.TablesLoading)
                return mEventWaitHandler.WaitOne(10000);
            return true;
        }

        public void onStatusChanged(O2GTableManagerStatus status, O2GTableManager manager)
        {
            mLastStatus = status;
            switch (status)
            {
                case O2GTableManagerStatus.TablesLoaded:
                    mLoaded = true;
                    mError = false;
                    mEventWaitHandler.Set();
                    break;
                case O2GTableManagerStatus.TablesLoadFailed:
                    mLoaded = false;
                    mError = true;
                    mEventWaitHandler.Set();
                    break;
            }
        }
    }
}

File: samples/macOS/java/TableManagerSamples/PrintTableListener/src/TableManagerListener.java
package printtablelistener;

import java.util.concurrent.Semaphore;
import java.util.concurrent.TimeUnit;

import com.fxcore2.*;

public class TableManagerListener implements IO2GTableManagerListener {
    O2GTableManagerStatus mLastStatus;
    private boolean mLoaded;
    private boolean mError;
    private Semaphore mSemaphore;

    public boolean isLoaded() {
        return mLoaded;
    }

    public boolean hasError() {
        return mError;
    }

    // ctor
    public TableManagerListener() {
        reset();
        mSemaphore = new Semaphore(0);
    }
    
    public void reset()
    {
        mLastStatus = O2GTableManagerStatus.TABLES_LOADING;
        mLoaded = false;
        mError = false;
    }

    public boolean waitEvents() throws Exception {
        if (mLastStatus == O2GTableManagerStatus.TABLES_LOADING)
            return mSemaphore.tryAcquire(30, TimeUnit.SECONDS);
        return true;
    }

    @Override
    public void onStatusChanged(O2GTableManagerStatus status, O2GTableManager manager) {
        mLastStatus = status;
        switch (status) {
        case TABLES_LOADED:
            mLoaded = true;
            mError = false;
            mSemaphore.release();
            break;
        case TABLES_LOAD_FAILED:
            mLoaded = false;
            mError = true;
            mSemaphore.release();
            break;
        }
    }
}
